lib/dotcom_web/components/live_components/trip_planner_results_section.ex

```diff
@@ -10,7 +10,10 @@ defmodule DotcomWeb.Components.LiveComponents.TripPlannerResultsSection do
 
   @impl true
   def mount(socket) do
-    {:ok, socket |> assign(:expanded_itinerary_index, nil)}
+    {:ok,
+     socket
+     |> assign(:expanded_itinerary_index, nil)
+     |> assign(:selected_itinerary_detail_index, 0)}
```

Comment:
Per [this comment](https://github.com/mbta/dotcom/pull/2238#issuecomment-2512450114), both of these bits of state are going to get moved up to the `TripPlanner` level.